src/custom_widgets/toast.rs

```diff
@@ -0,0 +1,469 @@
+use std::fmt;
+use std::time::{Duration, Instant};
+
+use iced::advanced::layout::{self, Layout};
+use iced::advanced::overlay;
+use iced::advanced::renderer;
+use iced::advanced::widget::{self, Operation, Tree};
+use iced::advanced::{Clipboard, Shell, Widget};
+use iced::event::{self, Event};
+use iced::mouse;
+use iced::theme;
+use iced::widget::{button, column, container, horizontal_rule, horizontal_space, row, text};
+use iced::window;
+use iced::{Alignment, Element, Length, Point, Rectangle, Renderer, Size, Theme, Vector};
+
+pub const DEFAULT_TIMEOUT: u64 = 5;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum Status {
+    #[default]
+    Primary,
+    Secondary,
+    Success,
+    Danger,
+}
+
+impl Status {
+    pub const ALL: &'static [Self] = &[Self::Primary, Self::Secondary, Self::Success, Self::Danger];
+}
+
+impl container::StyleSheet for Status {
+    type Style = Theme;
+
+    fn appearance(&self, theme: &Theme) -> container::Appearance {
+        let palette = theme.extended_palette();
+
+        let pair = match self {
+            Status::Primary => palette.primary.weak,
+            Status::Secondary => palette.secondary.weak,
+            Status::Success => palette.success.weak,
+            Status::Danger => palette.danger.weak,
+        };
+
+        container::Appearance {
+            background: Some(pair.color.into()),
+            text_color: pair.text.into(),
+            ..Default::default()
+        }
+    }
+}
+
+impl fmt::Display for Status {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Status::Primary => "Primary",
+            Status::Secondary => "Secondary",
+            Status::Success => "Success",
+            Status::Danger => "Danger",
+        }
+        .fmt(f)
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Toast {
+    pub title: String,
+    pub body: String,
+    pub status: Status,
+}
+
+pub struct Manager<'a, Message> {
+    content: Element<'a, Message>,
+    toasts: Vec<Element<'a, Message>>,
+    timeout_secs: u64,
```

Comment:
I guess a toast (Android style), does have a timeout and disapears on it's own.
Generally they don't have close buttons.

For the case of an about, that the user will want to read, do you think best for it to close on it's own?